### ISSUE TITLE
TextField: onRenderDescription is only invoked if description is present

### ISSUE THREAD

**Issue by chrismohr** [OfficeDev/office-ui-fabric-react#4669]
#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes,

Requested priority: Normal 

Products/sites affected: Yammer

#### Describe the issue:

On the TextField component, the `onRenderDescription` function is *only* called and rendered if the `description` or `errorMessage` props are truthy.

Our objective is to display jsx elements as the description on the TextField component by only providing `onRenderDescription`.  We are able to workaround the issue by providing a string to be ignored as the `description`. 

Relevant code:
https://github.com/OfficeDev/office-ui-fabric-react/blob/802a2c0847830acfbf47e2dd058da54d9dfb15ae/packages/office-ui-fabric-react/src/components/TextField/TextField.tsx#L193

#### Expected behavior:

`onRenderDescription` is called and rendered even when `description` or `errorMessage` are `undefined`.

### If applicable, please provide a codepen repro:

https://codepen.io/chrismohr/pen/jxqRga

**Comment by KevinTCoughlin** [comment]
Makes sense to me given that we seem to support customer description rendering intentionally. Definitely a legit bug, thanks for the Codepen repro @chrismohr!

One potential fix may be to expand the conditional here:

https://github.com/OfficeDev/office-ui-fabric-react/blob/802a2c0847830acfbf47e2dd058da54d9dfb15ae/packages/office-ui-fabric-react/src/components/TextField/TextField.tsx#L157

to depend on whether or not a custom description render function was supplied via props. I'd also keep an eye on the `aria-describedby` attributes to make sure they are properly rendered, but they should be.

Feel free to raise a PR otherwise I can probably get to it soon.

**Comment by chrismohr** [comment]
Thanks @KevinTCoughlin !
